infra/modules/service/main.tf

```diff
@@ -113,21 +113,33 @@ resource "aws_lb_target_group" "alb_target_group" {
 #######################
 
 resource "aws_ecs_service" "app" {
+  # checkov:skip=CKV_AWS_333: This is literally a new check; more research needed
+
+  # Fargate platform_version must be at least 1.4.0 for Fargate + EFS to work
+  # LATEST is currently 1.4.0
+  # See https://docs.aws.amazon.com/AmazonECS/latest/developerguide/platform-linux-fargate.html
   name                   = var.service_name
   cluster                = var.service_cluster_arn
   launch_type            = "FARGATE"
   task_definition        = aws_ecs_task_definition.app.arn
   desired_count          = var.desired_instance_count
-  platform_version       = "1.4.0"
+  platform_version       = "LATEST"
   enable_execute_command = var.enable_exec ? true : null
 
   # Allow changes to the desired_count without differences in terraform plan.
   # This allows autoscaling to manage the desired count for us.
-  # Ignoring the task_definition allows the task revision to not get reverted if a rew revision is created outside of terraform.
   lifecycle {
     ignore_changes = [
       desired_count,
-      # Comment out the following line to allow terraform to TEMPORARILY update the task definition.
+      # NOTE: By default, this module ignores changes to the ECS service task definition and ignores changes to the ECS task
+      # definition container definitions so that Github Actions can manage deploys by creating new task definition versions.
+      # Sometimes we need to deploy updates to container definitions and update the ECS service. Normally, we would control
+      # this behavior with a variable. However, terraform currently doesn't support expression evaluation in the `lifecycle`
+      # block. See https://github.com/hashicorp/terraform/issues/3116
+      #
+      # To deploy general updates to the task definition, **TEMPORARILY** uncomment the next line.
+      # To deploy general updates to the task definition and/or updates to the task's container definition,
+      #   **TEMPORARILY** uncomment the next line AND the `ignore_changes` line in the aws_ecs_task_definition below.
```

Comment:
I wonder whether this is better at the top of the file, once -- with a single line unique tag in a comment in the line above the `task_definition` and `container_definitions` lines.

Like:
> NOTE: By default..
> ... To change this, **TEMPORARILY** comment out the lines following `#  IGNORE_GITHUB_CHANGES - comment out to deploy changes via terraform` 

Or something? I dunno!

Reply:
Oh sure. I'll re-organize it. Thanks!